src/backend/distributed/executor/executor.c

```diff
@@ -1418,7 +1419,10 @@ RunDistributedExecution(DistributedExecution *execution)
 		 */
 		UnclaimAllSessionConnections(execution->sessionList);
 
-		pfree(events);
+		/*
+		 * We don't need to (and probably shouldn't) free events because
+		 * it might have never been allocated.
+		 */
```

Comment:
I don't think we need to explain that we don't do this, because I don't think the reader is necessarily expecting it. Freeing things is relatively unusual in postgres.